app/controllers/order/client_information.js

```diff
@@ -72,19 +87,60 @@ export default Ember.Controller.extend(cancelOrder, {
         actionType = "POST";
       }
 
-      var loadingView = getOwner(this).lookup('component:loading').append();
+      let loadingView = getOwner(this).lookup('component:loading').append();
 
-      new AjaxPromise(url, actionType, this.get('session.authToken'), { beneficiary: this.beneficiaryParams(), order_id: orderId })
+      if (clientInfo === 'organisation') {
+        let orderParams = {
+          'purpose_ids': purposeIds,
+          'beneficiary_id': null
+        };
+
+        new AjaxPromise('/orders/' + orderId, 'PUT', this.get('session.authToken'), { order: orderParams })
         .then(data => {
-          this.get("store").pushPayload(data);
-          loadingView.destroy();
-          if(this.get("previousRouteName") === "my_orders") {
-            this.get("myOrders").set("selectedOrder", this.get("order"));
-            this.transitionToRoute('my_orders');
-          } else {
-            this.transitionToRoute('order.goods_details', orderId, { queryParams: { fromClientInformation: true }});
+          this.store.pushPayload(data);
+          if (beneficiaryId) {
+            let beneficiary = this.store.peekRecord('beneficiary', beneficiaryId);
```

Comment:
Why we need to check in store if beneficiary exists or not? If Order has beneficiaryId then it simply means that beneficiary exists in th DB regardless whether it is in store or not..